dayFour/InputException.java

```diff
@@ -0,0 +1,56 @@
+package dayFour;
+
+abstract class InputHandler{
+	public abstract void handle(NegativeNumberException nne);
+	public abstract void handle(OutOfRangeException ore); 
+}
+
+class InputHandlerImpl extends InputHandler{
+	@Override
+	public void handle(NegativeNumberException nne) {
+		System.out.println("Enter a positive number...");
+	}
+	@Override
+	public void handle(OutOfRangeException ore) {
+		System.out.println("Enter a number between 1 and 100...");
+	}
+}
+
+public abstract class InputException extends Exception{
+	public InputException(String msg) {
+		super(msg);
+	}
+	public abstract void visit();
+}
+
+class NegativeNumberException extends InputException{
+	String msg;
+	public NegativeNumberException(String msg) {
+		super(msg);
+		this.msg=msg;
```

Comment:
Using shorthands (msg, nne, ore) for variable names is not a good practice. Prefer using camel-cased meaningful variable names going forward
`exceptionMessage`
`messageValue`

Reply:
I have modified the names to meaningful names and pushed it